library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2845,22 +2533,13 @@ impl<T: Clone, A: Allocator> VecDeque<T, A> {
 /// Returns the index in the underlying buffer for a given logical element index.
 #[inline]
 fn wrap_index(index: usize, size: usize) -> usize {
-    // size is always a power of 2
-    debug_assert!(size.is_power_of_two());
-    index & (size - 1)
-}
-
-/// Calculate the number of elements left to be read in the buffer
-#[inline]
-fn count(tail: usize, head: usize, size: usize) -> usize {
-    // size is always a power of 2
-    (head.wrapping_sub(tail)) & (size - 1)
+    if index >= size { index - size } else { index }
```

Comment:
suggestion: this only makes a difference in the CI runs, but I think this would be a good place for an extra debug_assert:
```suggestion
    debug_assert!(index < size || (index - size) < size);
    if index >= size { index - size } else { index }
```
just in case something in future accidentally adds some off-by-one mistakes or something.